### ISSUE TITLE
CANNOT reproduce the accuracy with OpenCLIP-L pretrained weights

### ISSUE THREAD

**Issue by Bin-NV** [VamosC/CLIP4STR#29]
hi,

I'm trying to reproduce your accuracy with OpenCLIP-ViT-L-14-DataComp-XL-s13B-b90K.bin, but after training with dataset str_dataset_ub,  the acc of all dataset is 0. could you pls share some instructions to debug?

Attached full training logs
[logs.txt](https://github.com/user-attachments/files/17742324/logs.txt)

**Comment by mzhaoshuai** [comment]
Hi, @Bin-NV 

A little bit weird...

After each epoch, there is a evaluation step
```
Epoch 0: 100%|██████████| 6330/6330 [1:38:13<00:00,  1.07it/s, v_num=.]
Epoch 0: 100%|██████████| 6330/6330 [1:38:13<00:00,  1.07it/s, v_num=.]
2024-11-13T15:52:43Z o 
Validation: |          | 0/? [00:00<?, ?it/s][A
2024-11-13T15:52:43Z o 
Validation:   0%|          | 0/8 [00:00<?, ?it/s][A
2024-11-13T15:52:43Z o 
Validation DataLoader 0:   0%|          | 0/8 [00:00<?, ?it/s][A
2024-11-13T15:52:43Z o 
Validation DataLoader 0:  12%|█▎        | 1/8 [00:00<00:02,  3.19it/s][A
2024-11-13T15:52:44Z o 
Validation DataLoader 0:  25%|██▌       | 2/8 [00:00<00:01,  3.20it/s][A
2024-11-13T15:52:44Z o 
Validation DataLoader 0:  38%|███▊      | 3/8 [00:00<00:01,  3.22it/s][A
2024-11-13T15:52:44Z o 
Validation DataLoader 0:  50%|█████     | 4/8 [00:01<00:01,  3.19it/s][A
2024-11-13T15:52:44Z o 
Validation DataLoader 0:  62%|██████▎   | 5/8 [00:01<00:00,  3.21it/s][A
2024-11-13T15:52:45Z o 
Validation DataLoader 0:  75%|███████▌  | 6/8 [00:01<00:00,  3.22it/s][A
2024-11-13T15:52:45Z o 
Validation DataLoader 0:  88%|████████▊ | 7/8 [00:02<00:00,  3.22it/s][A
2024-11-13T15:52:45Z o 
Validation DataLoader 0: 100%|██████████| 8/8 [00:02<00:00,  3.41it/s][A[2024-11-13 15:52:45,415][strhub.models.base][INFO] - binz-- val_accuracy 0.0 val_NED 0.43442691529214095 val_loss 3.6606194972991943
2024-11-13T15:52:45Z o 
2024-11-13T17:31:41Z o 
                                                                      [A
Epoch 0: 100%|██████████| 6330/6330 [1:38:17<00:00,  1.07it/s, v_num=.]
Epoch 0: 100%|██████████| 6330/6330 [1:38:17<00:00,  1.07it/s, v_num=.]
```

The val accuracy should be large than 90.0, rather than 0.0.

Beside, can u visualize the log with `tensorborad` to check the metrics like loss?

Is this only happens with `OpenCLIP-ViT-L-14-DataComp-XL-s13B-b90K.bin`? Will a `CLIP-ViT-B/16` model produce a similar result? I think you can use `CLIP-ViT-B/16` for a fast validation, for example, only 1-epoch training. It should also achieve a decent result.

I do not find the exact reason from the log now.

**Comment by Bin-NV** [comment]
I'm using torch2.2 and set `check_val_every_n_epoch=1` instead of `val_check_interval=1000` in trainer.

I tested the backbone like `OpenCLIP-ViT-L-14-DataComp-XL-s13B-b90K.bin` and `ViT-L-14.pt`, both got the 0.0 accuracy during validation

**Comment by mzhaoshuai** [comment]
Hi, @Bin-NV 
```
2024-11-13T22:46:56Z o 
2024-11-13T22:46:56Z o The execute file is /CLIP4STR_raw/code/CLIP4STR/test.py
2024-11-13T22:46:56Z o The best ckpt is /var/job/01eca3/output/vl4str_2024-11-13-14-12-27/checkpoints/epoch=0-step=792-val_accuracy=0.0000-val_NED=0.4174.ckpt
2024-11-13T22:46:56Z o The val accuracy is best 0.0-0e
2024-11-13T22:46:56Z o 
2024-11-13T22:46:56Z o  Test results with the last checkpoint
```

After training, it will use the best ckpt for test, and this best ckpt is choosed by the accuracy on evaluation set. The best ckpt is not updated after `step=792`. It maybe the first evaluation time.

Do you try a lower pytorch version like `1.13.0`?

**Comment by Bin-NV** [comment]
it works, thanks!

**Comment by mzhaoshuai** [comment]
> it works, thanks!

Happy to see this. If you solve the issue with a higher `pytorch` or `pytorch_lightning` version, please do consider raising a PR :smiley: :smiley: :smiley:.
